### PR TITLE
Revert to same version of System.CommandLine as 6.0.1xx.

### PR DESCRIPTION
**Pull request by dsplaisted**
Reverts #22351

### CODE REVIEW

File: eng/Versions.props [hunk 1]
```diff
@@ -36,7 +36,7 @@
     <SystemReflectionMetadataLoadContextVersion>6.0.0</SystemReflectionMetadataLoadContextVersion>
     <MicrosoftDeploymentDotNetReleasesVersion>1.0.0-preview1.1.21112.1</MicrosoftDeploymentDotNetReleasesVersion>
     <SystemManagementPackageVersion>4.6.0</SystemManagementPackageVersion>
-    <SystemCommandLineVersion>2.0.0-beta1.21601.2</SystemCommandLineVersion>
+    <SystemCommandLineVersion>2.0.0-beta1.21473.1</SystemCommandLineVersion>
```
Comment (MichaelSimons) [inline eng/Versions.props:39]: The command-line-api entries in Version.Details.xml need to be updated accordingly as well with this version.  Utilizing `darc update-dependencies` is a good way to ensure all entries are updated correctly.

**Comment by marcpopMSFT** [comment]
@dsplaisted I think templating needs to be on the same system.command line first as well. They are currently on 2.0.0-beta1.21525.1 in their 6.0.200 branch.

CC @MichaelSimons @vlada-shubina @baronfel

**Comment by marcpopMSFT** [comment]
@dsplaisted @MichaelSimons The darwin and source build legs have build failures installing 1.0.5. That is excluded from ubuntu with this line: https://github.com/dotnet/sdk/blob/32ce7ae9526e731d4dd254d828783255d10f2e1b/eng/restore-toolset.sh#L23

Did the underlying images change that those legs used such that they no longer support 1.0.5?

**Comment by vlada-shubina** [comment]
> @dsplaisted I think templating needs to be on the same system.command line first as well. They are currently on 2.0.0-beta1.21525.1 in their 6.0.200 branch.
> 
> CC @MichaelSimons @vlada-shubina @baronfel

I can revert templating 6.0.2xx branch to same version (2.0.0-beta1.21473.1)

File: src/Cli/dotnet/Parser.cs [hunk 2]
```diff
@@ -217,7 +190,7 @@ public override void Write(HelpContext context)
                 }
                 else if (command.Name.Equals(NewCommandParser.GetCommand().Name))
                 {
-                    NewCommandShim.Run(context.ParseResult.GetArguments());
+                    NewCommandShim.Run(helpArgs);
```
Comment (vlada-shubina) [inline src/Cli/dotnet/Parser.cs:193]: this fix is still needed. `dotnet new --help` and `dotnet new console --help` are different command.

**Comment by MichaelSimons** [comment]
> Did the underlying images change that those legs used such that they no longer support 1.0.5?

The docker images used in the SB leg has not changed.  This logic is busted for distros such as centos, fedora, etc.  Why is this check necessary?  If it is needed for certain scenarios, why not utilize an explicit setting to enable?

File: src/Cli/dotnet/Parser.cs (hunk 2 above)
Comment (baronfel) [reply src/Cli/dotnet/Parser.cs:193]: we dug into this some more - on its face this seems wrong for `dotnet new` (and any command with subcommands) but it turns out that in this revert the whole issue is sidestepped by the way the built-in commands are invoked in `Program.cs`. If a built-in command is detected then `Program.cs` forwards along the entire parseResult argument array to that command, which for `dotnet new` results in the correct help being returned, despite what this `if` block would suggest.  I've tested this branch locally to ensure this is the case, and we do get template-specific help when `dotnet.dll new console --help` is invoked.

Therefore, I suggest that we delete this `else if` entirely for clarity reasons on this 6.0.2xx branch. See the linked suggestion

Comment (vlada-shubina) [reply src/Cli/dotnet/Parser.cs:193]: We checked it more closely with @baronfel, due to other changes in Program.cs this code is not used anymore. Any dotnet new command is redirected to `NewCommandShim` as it is part of `BuiltInCommand`. 

However, the code is still invalid. I would recommend to remove this if branch as once this code becomes reachable the bug discovered in https://github.com/dotnet/sdk/issues/23162 will come back.

File: src/Cli/dotnet/Parser.cs [hunk 3]
```diff
@@ -217,7 +190,7 @@ public override void Write(HelpContext context)
...
                 }
```
Comment (baronfel) [inline src/Cli/dotnet/Parser.cs:194]: This whole range can be deleted due to https://github.com/dotnet/sdk/pull/23380/files#r783164794

```suggestion
```

**Comment by dsplaisted** [comment]
> > @dsplaisted I think templating needs to be on the same system.command line first as well. They are currently on 2.0.0-beta1.21525.1 in their 6.0.200 branch.
> > CC @MichaelSimons @vlada-shubina @baronfel
> 
> I can revert templating 6.0.2xx branch to same version (2.0.0-beta1.21473.1)

@vlada-shubina Yes, can you go ahead and do this?

**Comment by marcpopMSFT** [comment]
@dotnet/kitten Three of the test failures were accessviolations that may be coming from msbuild. Have ya'll seen reports of this?
Expected command to pass but it did not.\r\nFile Name: 
**1** 
D:\a\_work\1\s\artifacts\bin\redist\Debug\dotnet\dotnet.exe\r\nArguments: pack -o ../pkgs\r\nExit Code: -2146233082\r\nStdOut:\r\nMicrosoft (R) Build Engine version 17.1.0-preview-22063-01+f3c1fc9e5 for .NET\r\nCopyright (C) Microsoft Corporation. All rights reserved.\r\nStdErr:\r\nFatal error. System.AccessViolationException: Attempted to read or write protected memory. This is often an indication that other memory is corrupt.\r\n
**2**
Expected command to pass but it did not.\r\nFile Name: D:\a\_work\1\s\artifacts\bin\redist\Debug\dotnet\dotnet.exe\r\nArguments: add D:\a\_work\1\s\artifacts\tmp\Debug\WhenRefWithCo---4C8F2363\WithRefCondNonUniform\WithRefCondNonUniform.csproj reference -f net451 D:\a\_work\1\s\artifacts\tmp\Debug\WhenRefWithCo---4C8F2363\ValidRef\ValidRef.csproj\r\nExit Code: 57005\r\nStdOut:\r\n\r\nStdErr:\r\nFatal error. System.AccessViolationException: Attempted to read or write protected memory. This is often an indication that other memory is corrupt.\r\n   at Microsoft.Build.Framework.NativeMethods.IsLongPathsEnabledRegistry()\r\n   at Microsoft.Build.Framework.NativeMethods.IsMaxPathLegacyWindows()

**3**
Expected command to pass but it did not.\r\nFile Name: D:\a\_work\1\s\artifacts\bin\redist\Debug\dotnet\dotnet.exe\r\nArguments: build --configuration Debug\r\nExit Code: 57005\r\nStdOut:\r\nMicrosoft (R) Build Engine version 17.1.0-preview-22063-01+f3c1fc9e5 for .NET\r\nCopyright (C) Microsoft Corporation. All rights reserved.\r\n\r\n  Determining projects to restore...\r\n  Restored D:\a\_work\1\s\artifacts\tmp\Debug\ItPassesDepsf---D4353578\TestAppWithProjDepTool.csproj (in 1.04 sec).\r\n  Restored D:\a\_work\1\s\artifacts\tmp\Debug\ItPassesDepsf---D4353578\TestAppWithProjDepTool.csproj (in 1.19 sec).\r\n  You are using a preview version of .NET. See: https://aka.ms/dotnet-core-preview\r\nStdErr:\r\nFatal error. System.AccessViolationException: Attempted to read or write protected memory. This is often an indication that other memory is corrupt.\r\n   at System.Reflection.Metadata.MetadataReader.ReadMetadataTableRowCounts(System.Reflection.Metadata.BlobReader ByRef, UInt64)\r\n   at System.Reflection.Metadata.MetadataReader.ReadMetadataTableHeader(System.Reflection.Metadata.BlobReader ByRef, System.Reflection.Metadata.Ecma335.HeapSizes ByRef, Int32[] ByRef, System.Reflection.Metadata.Ecma335.TableMask ByRef)\r\n   at System.Reflection.Metadata.MetadataReader..ctor(Byte*, Int32, System.Reflection.Metadata.MetadataReaderOptions, System.Reflection.Metadata.MetadataStringDecoder, System.Object)\r\n   at System.Reflection.Metadata.PEReaderExtensions.GetMetadataReader(System.Reflection.PortableExecutable.PEReader, System.Reflection.Metadata.MetadataReaderOptions, System.Reflection.Metadata.MetadataStringDecoder)\r\n   at System.Reflection.Metadata.PEReaderExtensions.GetMetadataReader(System.Reflection.PortableExecutable.PEReader)\r\n   at Microsoft.Build.Shared.AssemblyNameExtension.GetAssemblyNameEx(System.String)

**Comment by Forgind** [comment]
We actually had an AV come up recently, and we suspected it was antivirus software. Is there a dump of the problem somewhere?

**Comment by Forgind** [comment]
It should be almost impossible to get that from managed code.

**Comment by wtgodbe** [comment]
@dsplaisted is this good to go?

**Comment by wtgodbe** [comment]
Ah, Daniel's out - @marcpopMSFT were your concerns fixed by the last commit?

**Comment by marcpopMSFT** [comment]
@wtgodbe merged since it passed but the msbuild folks should be aware that there may be some instability if this happens again.

**Pull request merged**